azurerm/internal/services/sentinel/resource_arm_sentinel_alert_rule_action.go

```diff
@@ -0,0 +1,168 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmSentinelAlertRuleAction() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSentinelAlertRuleActionCreate,
+		Read:   resourceArmSentinelAlertRuleActionRead,
+		Delete: resourceArmSentinelAlertRuleActionDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.SentinelAlertRuleActionID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"rule_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SentinelAlertRuleID,
+			},
+
+			"logic_app_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: logicValidate.LogicAppWorkflowID,
+			},
+		},
+	}
+}
+
+func resourceArmSentinelAlertRuleActionCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sentinel.AlertRulesClient
+	logicTriggerClient := meta.(*clients.Client).Logic.WorkflowTriggersClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+
+	ruleID, err := parse.SentinelAlertRuleID(d.Get("rule_id").(string))
+	if err != nil {
+		return err
+	}
+	lappId, err := logicParse.LogicAppWorkflowID(d.Get("logic_app_id").(string))
+	if err != nil {
+		return err
+	}
+
+	// Ensure no existed resources
+	resp, err := client.GetAction(ctx, ruleID.ResourceGroup, ruleID.Workspace, ruleID.Name, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("checking for existing Sentinel Alert Rule Action %q (Resource Group %q / Workspace %q / Rule %q): %+v", name, ruleID.ResourceGroup, ruleID.Workspace, ruleID.Name, err)
+		}
+	}
+
+	if resp.ID != nil && *resp.ID != "" {
+		return tf.ImportAsExistsError("azurerm_sentinel_alert_rule_action", *resp.ID)
+	}
+
+	// List callback URL for sentinel alert specific trigger from the workspace containing specified alert rule.
+	triggerName := "When_a_response_to_an_Azure_Sentinel_alert_is_triggered"
+	tresp, err := logicTriggerClient.ListCallbackURL(ctx, ruleID.ResourceGroup, lappId.Name, triggerName)
+	if err != nil {
+		return fmt.Errorf("listing callback URL for Logic App Trigger %q (Resource Group %q / Workspace %q): %+v", triggerName, ruleID.ResourceGroup, ruleID.Workspace, err)
+	}
```

Comment:
is it likely these could change in time/should be passed in?

Reply:
This is exactly what the portal does. I feel user do not necessary need to care about the callback URL here.
On the other hand, if we want to pass it in, then do you think it make sense to expose a `Computed` field in `azurerm_logic_app_trigger_custom` and refer to it here?

Reply:
Another way is that we specify anther field `logic_app_trigger_name`, meanwhile submit an issue to azure api spec to ask them make the API idempotent.

Reply:
Azure/azure-rest-api-specs#9424 is opened for this.